Fix for compiling PHP 5.3.10 on Ubuntu 14.10.

Patch was taken from: https://mail.gnome.org/archives/xml/2012-August/msg00028.html

The patch was only slightly modified to work with patch parameter "-p1":
  sed -i -e "s|--- ext|--- a/ext|" php-5.4.5-libxm2-2.9.0.patch 
  sed -i -e "s|+++ ext|+++ b/ext|" php-5.4.5-libxm2-2.9.0.patch

For reference: When the error occures, php-build shows the following log:
                                                                  ^
/tmp/php-build/source/5.3.10/Zend/zend_API.h:547:32: note: in definition of macro ‘ZVAL_STRINGL’
   const char *__s=(s); int __l=l;  \
                                ^
/tmp/php-build/source/5.3.10/ext/simplexml/simplexml.c:1390:4: note: in expansion of macro ‘RETVAL_STRINGL’
    RETVAL_STRINGL((char *)outbuf->buffer->content, outbuf->buffer->use, 1);
    ^
make: *** [ext/simplexml/simplexml.lo] Error 1
make: *** Waiting for unfinished jobs....


--- a/ext/dom/node.c	2012-08-06 17:49:48.826716692 +0800
+++ b/ext/dom/node.c	2012-08-06 17:52:47.633484660 +0800
@@ -1895,9 +1895,17 @@ static void dom_canonicalization(INTERNA
         RETVAL_FALSE;
     } else {
 		if (mode == 0) {
+#ifdef LIBXML2_NEW_BUFFER
+            ret = xmlOutputBufferGetSize(buf);
+#else
 			ret = buf->buffer->use;
+#endif
 			if (ret > 0) {
+#ifdef LIBXML2_NEW_BUFFER
+                RETVAL_STRINGL((char *) xmlOutputBufferGetContent(buf), ret, 1);
+#else
 				RETVAL_STRINGL((char *) buf->buffer->content, ret, 1);
+#endif
 			} else {
 				RETVAL_EMPTY_STRING();
 			}
--- a/ext/dom/documenttype.c	2012-08-06 18:02:16.019640870 +0800
+++ b/ext/dom/documenttype.c	2012-08-06 18:06:16.612228905 +0800
@@ -205,7 +205,13 @@ int dom_documenttype_internal_subset_rea
 		if (buff != NULL) {
 			xmlNodeDumpOutput (buff, NULL, (xmlNodePtr) intsubset, 0, 0, NULL);
 			xmlOutputBufferFlush(buff);
+
+#ifdef LIBXML2_NEW_BUFFER
+			ZVAL_STRINGL(*retval, xmlOutputBufferGetContent(buff),
+			             xmlOutputBufferGetSize(buff), 1);
+#else
 			ZVAL_STRINGL(*retval, buff->buffer->content, buff->buffer->use, 1);
+#endif
 			(void)xmlOutputBufferClose(buff);
 			return SUCCESS;
 		}
--- a/ext/simplexml/simplexml.c	2012-08-06 18:10:44.621017026 +0800
+++ b/ext/simplexml/simplexml.c	2012-08-06 18:12:48.016270419 +0800
@@ -1417,7 +1417,12 @@ SXE_METHOD(asXML)
 
 			xmlNodeDumpOutput(outbuf, (xmlDocPtr) sxe->document->ptr, node, 0, 0, ((xmlDocPtr) sxe->document->ptr)->encoding);
 			xmlOutputBufferFlush(outbuf);
+#ifdef LIBXML2_NEW_BUFFER
+			RETVAL_STRINGL((char *)xmlOutputBufferGetContent(outbuf),
+			               xmlOutputBufferGetSize(outbuf), 1);
+#else
 			RETVAL_STRINGL((char *)outbuf->buffer->content, outbuf->buffer->use, 1);
+#endif
 			xmlOutputBufferClose(outbuf);
 		}
 	} else {
